lib/smart_proxy_ansible/runner/ansible_runner.rb

```diff
@@ -28,6 +28,9 @@ def initialize(input, suspended_action:, id: nil)
         @passphrase = action_input['secrets']['key_passphrase']
         @execution_timeout_interval = action_input[:execution_timeout_interval]
         @cleanup_working_dirs = action_input.fetch(:cleanup_working_dirs, true)
+        @host = action_input.fetch(:hostname)
```

Comment:
Additionally, things get a little complicated with ansible, because a host can have `ansible_host` parameter set which will effectively override the name in the inventory, although I'm not sure how common that is.

Reply:
I didn't observe the behavior you're describing. If a host has an `ansible_host` parameter, it overrides the `ansible_host` value, not the `hostname` value.

Reply:
If `ansible_host` parameter is unset, then ansible uses the inventory name to connect to. The same name will be used during the known hosts cleanup here so everything is fine.

If however ansible_host is set and points to a different name than the inventory name, then we could get into a situation where the cleanup will be done for one hostname, but ansible will then connect using a different one. This mismatch could eventually turn out to be problematic

Reply:
Thanks for the clarification. We can check both `hostname` and `ansible_host` to resolve this.

Reply:
Since we're considering the `ansible_host` parameter, I thought it made sense to include `ansible_port` as well. This does make the code slightly more complex, but I believe it improves consistency. Let me know what you think.

Reply:
Yes, that's a good idea